### ISSUE TITLE
Scheduler task never executes and no `CallUnavailable` is emitted if Preimage is missing at execution time

### ISSUE THREAD

**Issue by wilwade** [paritytech/substrate#13980]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

The scheduler has a pair of edge cases that combine to cause invalid state as well as failure to notify.

- Found in v0.9.36 (may exist in prior)
- Still exists currently

If a lookup based call is scheduled, but the preimage is removed before execution, no failure event is deposited nor is the event added to `incompleteSince` or removed from `agenda` state.

https://github.com/paritytech/substrate/blob/c7d26dcf3d6c19fa7dcbd1f2633e337e5a5cbc40/frame/scheduler/src/lib.rs#L1065-L1068

Occurs prior to to place where the deposit event occurs inside of `service_task`

https://github.com/paritytech/substrate/blob/c7d26dcf3d6c19fa7dcbd1f2633e337e5a5cbc40/frame/scheduler/src/lib.rs#L1076-L1083

This causes a lack of a `deposit_event` for the call.

The `service_agenda` call then fails to consider `Unavailable` as `postponed` and will return true which results in state that will never be removed.

https://github.com/paritytech/substrate/blob/c7d26dcf3d6c19fa7dcbd1f2633e337e5a5cbc40/frame/scheduler/src/lib.rs#L1022-L1044

Either these calls should be considered postponed, or the state for them should not be added back into the `Agenda` storage.

### Steps to reproduce

1. Create a preimage for a referenda to execute
2. Create and pass the referenda for said preimage
3. Current State: Preimage (exists and requested) ✅ , scheduler: scheduled ✅ 
4. Now unrequest and unnote the preimage
5. Current State: Preimage: Empty, scheduler: still scheduled ✅ 
6. Continue forming blocks until the scheduler attempts to execute scheduled task
7. Expected: `CallUnavailable` event emitted. Actual: No block events

Resulting Error State:
- `scheduler.incompleteSince` state: Empty
- `scheduler.agenda` state: Contains prior block agenda item (the scheduled task)

Expected Error state:
- Option 1: `incompleteSince` contains task waiting for preimage
- Option 2: `agenda` is empty

**Comment by bkchr** [comment]
> 4\. Now unrequest and unnote the preimage

How does this happen?

**Comment by wilwade** [comment]
@bkchr You are correct that it cannot happen without either changing the preimage pallet `ManagerOrigin` say to a collective member or via sudo powers (or I suppose a fast track that does it). The easy way to recreate is to do step 4 with sudo powers.
